### PR TITLE
Interactive matplotlib-based ligand atom mapping visualization

### PR DESCRIPTION
**Pull request by dwhswenson**
* `network_plotting.py`: Generic tools for interactive matplotlib plots of networks. Includes customizable event handling (selecting edges, selecting nodes, dragging nodes). This can be reused for other kinds of networks (example: we could give a pop-up of parameters for how an edge will be calculated).
* `ligand_network_plotting.py`: Subclasses of things in `network_plotting` that are specific to plotting atom mappings in a ligand network.

Putting this up now; still needs tests and better docstrings.

---

To-do items for this:

- [x] Add partial transparency to drawings, so we can see the graph behind them
- [x] Hide axes ticks

These items have been moved to issues

- Change nodes to text labels, using string name of molecule
- Add bounding box around the drawings
- Show unhighlighted molecule on node hover or node click
- Blitting (for performance)
- Cache edges for each node on mousedown (might help performance on dragging)
- Add support for multigraph (multiple distinct edges for the same pair of nodes)

### CODE REVIEW

**Comment by dwhswenson** [comment]
Requesting review at this point; I'm still hunting the last corners of test coverage, etc., but this is a huge PR, so I'd rather you all start reviewing sooner than later.

Things on the to-do list that are not in this PR are going to issues for backlog.

**Review by IAlibay — APPROVED** [review]
Just the one thing, otherwise lgtm. As mentioned privately, I have a feeling you've had a much deeper dive into interactive matplotlib plotting than any of us, so "it works" is probably the main outcome that needed to be passed here.

File: openfe/utils/atommapping_network_plotting.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        # version A: using AxesImage
+        im = matplotlib.image.AxesImage(ax, extent=extent, zorder=10)
+
+        # version B: using BboxImage
+        # keep this commented code around for later performance checks
+        # bounds = (x0, y0, x1 - x0, y1 - y0)
+        # bounds = (0.2, 0.2, 0.3, 0.3)
+        # bbox0 = matplotlib.transforms.Bbox.from_bounds(*bounds)
+        # bbox = matplotlib.transforms.TransformedBbox(bbox0, ax.transAxes)
+        # im = matplotlib.image.BboxImage(bbox)
```
[the remark this answers is not in the document]
Comment (IAlibay) [reply openfe/utils/atommapping_network_plotting.py:62]: pinging @dwhswenson - we can merge after this.

[the remark this answers is not in the document]
Comment (dwhswenson) [reply openfe/utils/atommapping_network_plotting.py:62]: Sorry -- yep, it's meant to be there. Took me a while to figure out the other version; I didn't play with performance yet to see if it make a difference (it looks like more code, but who knows what's going on under the hood with matplotlib)

[the remark this answers is not in the document]
Comment (IAlibay) [reply openfe/utils/atommapping_network_plotting.py:62]: Cool thanks.

**Pull request merged**